src/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -50,42 +52,44 @@ const DiffExprCompute = (props) => {
     const numMetadata = Object.values(properties).filter((o) => o.type === 'metadataCategorical').length;
 
     if (!numMetadata) {
-      setType(ComparisonType.within);
+      dispatch(setComparisonType(ComparisonType.within));
     }
     setHasMetadata(numMetadata > 0);
 
-    // Update the hierarchies if a previously selected set were to be deleted.
-    setSelectedGroups(_.mapValues(selectedGroups, (cellSetKey) => {
-      if (!cellSetKey) {
-        return null;
-      }
-
-      const splitKey = cellSetKey.split('/')[1];
+    // If any key in group option is deleted, set option to null
+    Object.keys(comparisonGroup).forEach((type) => {
+      const groupDiff = {};
```

Comment:
do we have a unit test for this case?

Reply:
also, this code is quite hard to read and understand as it is. I would first extract the "check if a key in group option has deleted" part as a separate function. Then, in the use effect I would just have an if statement checking if that function is true, and if it is true I would proceed with setting the key to null

Reply:
Removed, as not relevant anymore per discussion with Vicky regarding in-group comparison